x-pack/plugins/apm/server/lib/traces/get_top_traces.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SearchResponse } from 'elasticsearch';
+import { get } from 'lodash';
+import { PARENT_ID, TRACE_ID } from '../../../common/constants';
+import { StringMap } from '../../../typings/common';
+import { Transaction } from '../../../typings/Transaction';
+import { ITransactionGroup } from '../../../typings/TransactionGroup';
+import {
+  prepareTransactionGroups,
+  TRANSACTION_GROUP_AGGREGATES
+} from '../helpers/transaction_group_query';
+
+export async function getTopTraces({
+  setup
+}: {
+  setup: StringMap<any>;
+}): Promise<ITransactionGroup[]> {
+  const { start, end, esFilterQuery, client, config } = setup;
+
+  const params = {
+    index: config.get('apm_oss.transactionIndices'),
+    body: {
+      size: 0,
+      query: {
+        bool: {
+          must: {
+            // this criterion safeguards against data that lacks a transaction
+            // parent ID but still is not a "trace" by way of not having a
+            // trace ID (e.g. old data before parent ID was implemented, etc)
+            exists: {
+              field: TRACE_ID
+            }
+          },
```

Comment:
Errors can also have `trace_id`, so we should probably make sure it's a transaction:
`{ term: { [PROCESSOR_EVENT]: 'transaction' } }`

Reply:
Oh ok. Small confusion here: we don't currently do this same check in `get_top_transactions` (we seem to check for a single transaction type instead, perhaps?) and the get_error_groups query searches on a different index `config.get('apm_oss.errorIndices')`. What am I missing here and what's `PROCESSOR_EVENT`?

Reply:
> we don't currently do this same check in get_top_transactions (we seem to check for a single transaction type instead, perhaps?) 

Yes, error don't have `transaction.type` field. But to make it obvious we should probably add `processor.event` there as well.

> get_error_groups query searches on a different index config.get('apm_oss.errorIndices')

The default index is `apm-*`:
https://github.com/elastic/kibana/blob/master/src/core_plugins/apm_oss/index.js#L33

Which is why we add:`
`{ term: { [PROCESSOR_EVENT]: 'error' } }`

Reply:
> What am I missing here and what's PROCESSOR_EVENT

Processor event is basically the document type. Valid types are `transaction`, `span` and `error`